arch/riscv/src/syscall.rs

```diff
@@ -47,11 +78,16 @@ const R_A3: usize = 12;
 const R_A4: usize = 13;
 
 /// Values for encoding the stored state buffer in a binary slice.
-const VERSION: u32 = 1;
-const STORED_STATE_SIZE: u32 = size_of::<Riscv32iStoredState>() as u32;
+const VERSION: usize = 1;
+const STORED_STATE_SIZE: usize = size_of::<RiscvStoredState>();
+#[cfg(target_arch = "riscv32")]
 const TAG: [u8; 4] = [b'r', b'v', b'5', b'i'];
+#[cfg(any(target_arch = "riscv64", not(target_os = "none")))]
+const TAG: [u8; 8] = [b'r', b'v', b'5', b'i', b'r', b'v', b'5', b'i'];
 const METADATA_LEN: usize = 3;
 
+// TODO: CHERI. This seems to be for swap or some such. Needs thinking about.
```

Comment:
?

Reply:
Yeah, I was not sure what a lot of the code was for and so couldn't quite work out what needed to be done. It seemed to me to just be serialising state.

I can just remove this comment, but maybe I could also do a useful CHERI implementation if I knew what the purpose was?

Reply:
Removed